kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -369,8 +389,31 @@ public LogSegment getLogSegmentForVersion(Engine engine, Optional<Long> versionT
 
     logDebugFileStatuses("deltasAfterCheckpoint", deltasAfterCheckpoint);
 
+    //////////////////////////////////////////////////////////////////////////////////
+    // Step 8: Grab all compactions in range [$latestCompleteCheckpointVersion + 1, //
+    //         $versionToLoad]                                                      //
+    //////////////////////////////////////////////////////////////////////////////////
+
+    final List<FileStatus> compactionsAfterCheckpoint;
```

Comment:
nit: Can we remove the `if (USE_COMPACTED_FILES) {` ?

If USE_COMPACTED_FILES is false, then above we would have set `listedCompactionFileStatuses` to `Collections.emptyList()`

And then here, `.stream.filter.collect` would just do nothing, right?